src/app/services/auth.service.ts

```diff
@@ -18,10 +18,10 @@ export class AuthService {
       if (user) {
         this.userData = new User(user as IUser);
         localStorage.setItem('user', JSON.stringify(this.userData));
-        JSON.parse(localStorage.getItem('user') || '{}');
+        JSON.parse(localStorage.getItem('user') || 'null');
       } else {
-        localStorage.setItem('user', null || '{}');
-        JSON.parse(localStorage.getItem('user') || '{}');
+        localStorage.setItem('user', 'null');
+        JSON.parse(localStorage.getItem('user') || 'null');
```

Comment:
What happens if you dont use 'null' here?

Reply:
it is giving an error because localStorage getItem returns string or null it gives strict error so to handle null condition I kept null string